opentelemetry-datadog/src/lib.rs

```diff
@@ -0,0 +1,123 @@
+//! ```
+//!
+//! [`tokio-support`]: https://tokio.rs
+//! [`async-std`]: https://async.rs
+//!
+
+//! ## Bring your own http client
+//!
+//! Users can choose appropriate http clients to align with their runtime.
+//!
+//! Based on the feature enabled. The default http client will be different. If user doesn't specific
+//! features or enabled `reqwest-blocking-client` feature. The blocking reqwest http client will be used as
+//! default client. If `reqwest-client` feature is enabled. The async reqwest http client will be used. If
+//! `surf-client` feature is enabled. The surf http client will be used.
+//!
+//! Note that async http clients may need specific runtime otherwise it will panic. User should make
+//! sure the http client is running in appropriate runime.
+//!
+//! Users can always use their own http clients by implementing `HttpClient` trait.
+//!
+//! ## Kitchen Sink Full Configuration
+//!
+//! Example showing how to override all configuration options. See the
+//! [`DatadogPipelineBuilder`] docs for details of each option.
+//!
+//! [`DatadogPipelineBuilder`]: struct.DatadogPipelineBuilder.html
+//!
+//! ```no_run
+//! use opentelemetry::{KeyValue, trace::Tracer};
+//! use opentelemetry::sdk::{trace::{self, IdGenerator, Sampler}, Resource};
+//! use opentelemetry::sdk::export::trace::ExportResult;
+//! use opentelemetry_datadog::{new_pipeline, ApiVersion, Error};
+//! use opentelemetry_http::HttpClient;
+//! use async_trait::async_trait;
+//!
+//! // `reqwest` and `surf` are supported through features, if you prefer an
+//! // alternate http client you can add support by implementing `HttpClient` as
+//! // shown here.
+//! #[derive(Debug)]
+//! struct IsahcClient(isahc::HttpClient);
+//!
+//! #[async_trait]
+//! impl HttpClient for IsahcClient {
+//!   async fn send(&self, request: http::Request<Vec<u8>>) -> ExportResult {
+//!     let result = self.0.send_async(request).await.map_err(|err| Error::Other(err.to_string()))?;
+//!
+//!     if result.status().is_success() {
+//!       Ok(())
+//!     } else {
+//!       Err(Error::Other(result.status().to_string()).into())
+//!     }
+//!   }
+//! }
+//!
+//! fn main() -> Result<(), opentelemetry::trace::TraceError> {
+//!     let (tracer, _uninstall) = new_pipeline()
+//!         .with_service_name("my_app")
+//!         .with_version(ApiVersion::Version05)
+//!         .with_agent_endpoint("http://localhost:8126")
+//!         .with_trace_config(
+//!             trace::config()
+//!                 .with_default_sampler(Sampler::AlwaysOn)
+//!                 .with_id_generator(IdGenerator::default())
+//!         )
+//!         .install()?;
+//!
+//!     tracer.in_span("doing_work", |cx| {
+//!         // Traced app logic here...
+//!     });
+//!
+//!     Ok(())
+//! }
+//! ```
+
+mod exporter;
+mod propagator;
+
+pub use exporter::{new_pipeline, ApiVersion, DatadogExporter, DatadogPipelineBuilder, Error};
+pub use propagator::DatadogPropagator;
```

Comment:
Same here, I'd remove some layers and/or use inline modules unless there's actually substantial code in a module.